include/pops/treatments.hpp

```diff
@@ -175,29 +176,40 @@ class SimpleTreatment : public BaseTreatment<IntegerRaster, FloatRaster>
         IntegerRaster& infected,
         std::vector<IntegerRaster>& exposed,
         IntegerRaster& susceptible,
-        IntegerRaster&,
+        IntegerRaster& resistant,
+        IntegerRaster& total_hosts,
         const std::vector<std::vector<int>>& suitable_cells) override
     {
         for (auto indices : suitable_cells) {
             int i = indices[0];
             int j = indices[1];
+            int exposed_changed = 0;
+            int infected_changed = 0;
+            int susceptible_changed = 0;
             if (this->application_ == TreatmentApplication::Ratio) {
-                infected(i, j) = infected(i, j) - (infected(i, j) * this->map_(i, j));
+                infected_changed = infected(i, j) - (infected(i, j) * this->map_(i, j));
+                infected(i, j) = infected_changed;
             }
             else if (this->application_ == TreatmentApplication::AllInfectedInCell) {
-                infected(i, j) = this->map_(i, j) ? 0 : infected(i, j);
+                infected_changed = this->map_(i, j) ? 0 : infected(i, j);
+                infected(i, j) = infected_changed;
             }
             for (auto& raster : exposed) {
                 if (this->application_ == TreatmentApplication::Ratio) {
                     raster(i, j) = raster(i, j) - (raster(i, j) * this->map_(i, j));
+                    exposed_changed += raster(i, j) - (raster(i, j) * this->map_(i, j));
```

Comment:
Here you need a new local variable, something like `new_exposed_individual` or rather a simple `new_value`, and 1) assign it to `raster(i, j)` and add it to the new exposed total.